### ISSUE TITLE
no-fallthrough does not allow comments between empty cases

### ISSUE THREAD

**Issue by jankuca** [eslint/eslint#7025]
Multiple cases with a single body are always allowed but placing
comments between the cases is not allowed.

``` js
switch(foo) {
case 1:
case 2:
doSomething();
}

switch(foo) {
// We need to handle 1.
case 1:
// We also need to handle 2.
case 2:
doSomething();
}
```

The latter is not valid as the comment between `case 1` and `case 2` is not allowed. Would be great if it was possible to comment on individual cases without having to add the "fallthrough" comment after that.

**Comment by eslintbot** [comment]
Hi @jankuca, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Is this a rule enhancement request or a bug report? What is "this option" that is set to `true`? Please add some details.

**Comment by jankuca** [comment]
@platinumazure I updated the description, sorry. The original description was referring to the PR I created (#7026).

The issue is that comments placed between cases with no body (i.e. cases which are handled the same way as the next case) are not accepted by the `no-fallthrough` rule. Here is an example of what is not accepted by the rule:

```js
switch(foo) {
// We need to handle 1.
case 1:
// We also need to handle 2. => This is not allowed without the fallthrough comment.
case 2:
doSomething();
}
```

It currently has to be written as the following. The proposal is to add an option which would not require the fallthrough comment for situations in which there is no code between switch cases.

```js
switch(foo) {
// We need to handle 1.
case 1:
// We also need to handle 2.
// fallthrough => This is ok but rather verbose since there is no code between case 1 and case 2.
case 2:
doSomething();
}
```

**Comment by vitorbal** [comment]
@platinumazure looks like we lost track of this one. @jankuca thanks for getting back to us. You could add "fallthrough" to the beginning of your comment like this:
```javascript
switch (foo) {
	case 1:
	// fallthrough: handled the same as 2.
	case 2:
    doSomething();
}
```

Would that be enough to solve your issue?

**Comment by vitorbal** [comment]
friendly ping @jankuca, could the suggestion above be an acceptable solution for what you described?

**Comment by kaicataldo** [comment]
@jankuca One more friendly ping

**Comment by jankuca** [comment]
Hi, the solution you are suggesting is what I was objecting against in the first place.

**Comment by kaicataldo** [comment]
@jankuca: @vitorbal's is suggesting that you can combine the `fall through` comment with the other comment.

```js
switch(foo) {
// We need to handle 1.
case 1:
// fallthrough: We also need to handle 2.
case 2:
doSomething();
}
```

**Comment by kaicataldo** [comment]
There's also the `commentPattern`: option: http://eslint.org/docs/rules/no-fallthrough#commentpattern

If you don't want this behavior, wouldn't it best to just turn the rule off? It sounds like what you're trying to enforce is having ANY comment between cases make the rule consider the fall through  case as valid, which is error prone and seems to me to defeat the purpose of enabling the rule.

**Comment by vitorbal** [comment]
Agree with @kaicataldo here. Not sure I see much additional value in this new option, considering what's already possible with `commentPattern`. I am :-1: for this change.

**Comment by not-an-aardvark** [comment]
It looks like there hasn't been any activity on this issue for awhile, and there are two :-1:s from team members, so I'm going to go ahead and close it. Of course, you can always implement this behavior yourself as part of a custom rule.
